src/main/java/com/ldtteam/structurize/storage/StructurePacks.java

```diff
@@ -0,0 +1,614 @@
+package com.ldtteam.structurize.storage;
+
+import com.google.gson.internal.Streams;
+import com.google.gson.stream.JsonReader;
+import com.ldtteam.structurize.api.util.Log;
+import com.ldtteam.structurize.blueprints.v1.Blueprint;
+import com.ldtteam.structurize.blueprints.v1.BlueprintUtil;
+import net.minecraft.Util;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NbtIo;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import java.util.function.Predicate;
+
+/**
+ * Class that contains all the structurepacks of the instance.
+ */
+public class StructurePacks
+{
+    /*
+     * todo add md5 support in the future (stronger consistency guarantee).
+     * This way we can be 100% sure that the pack has the correct version.
+     */
+
+    /**
+     * Current pack format.
+     * Increase when the pack data format changes, or minecraft version changes require a full schematic update.
+     * When the pack format doesn't align, the pack won't be loaded.
+     */
+    public static final int PACK_FORMAT = 1;
+
+    /**
+     * The list of registered structure packs.
+     * This might be accessed concurrently by client/server. That's why it is a concurrent hashmap.
+     */
+    public static Map<String, StructurePackMeta> packMetas = new ConcurrentHashMap<>();
+
+    /**
+     * Set to true on client/server once style loading has finished.
+     */
+    public static volatile boolean finishedLoading = false;
+
+    /**
+     * Selected pack on the client.
+     */
+    public static StructurePackMeta selectedPack;
+
+    /**
+     * Get a blueprint future.
+     * @param structurePackId the structure pack the blueprint is in.
+     * @param subPath the path of the specific blueprint in the pack.
+     * @return the blueprint future (might contain null).
+     */
+    public static Future<Blueprint> getBlueprintFuture(final String structurePackId, final String subPath)
+    {
+        return Util.ioPool().submit(() -> getBlueprint(structurePackId, subPath));
```

Comment:
Lets just use our own pool here.
I think that is more resilient for the future.